pkg/edgeview/src/network.go

```diff
@@ -0,0 +1,1256 @@
+				niType[niStatus.Type], niStatus.Logicallabel)
+			fmt.Printf(" DHCP range start: %v, end: %v\n", niStatus.DhcpRange.Start, niStatus.DhcpRange.End)
+			fmt.Printf(" Current Uplink: %s\n", niStatus.CurrentUplinkIntf)
+			fmt.Printf(" Probe Status:\n")
+			for k, p := range niStatus.PInfo {
+				fmt.Printf(" Uplink Intfname: %s\n", k)
+				upStatus := "Down"
+				if p.SuccessCnt != 0 || p.SuccessProbeCnt != 0 {
+					upStatus = "UP"
+				}
+				fmt.Printf("   Probe status: %s, Cost: %d, local success: %d, remote success: %d\n",
+					upStatus, p.Cost, p.SuccessCnt, p.SuccessProbeCnt)
+			}
+			fmt.Printf("\n")
+		}
+		closePipe(true)
+	}
+
+	appUUIDStr := appStatus.UUIDandVersion.UUID.String()
+	retbytes, err := ioutil.ReadFile("/run/domainmgr/DomainStatus/"+appUUIDStr+".json")
+	if err == nil {
+		printColor("\n  - domain status:", colorGREEN)
+		var domainS types.DomainStatus
+		_ = json.Unmarshal(retbytes, &domainS)
+		fmt.Printf("    state: %d, boot time: %v, tried count %d\n",
+			domainS.State, domainS.BootTime, domainS.TriedCount)
+		if domainS.Error != "" {
+			fmt.Printf("    error: %s, error time: %v, boot failed: %v",
+				domainS.Error, domainS.ErrorTime, domainS.BootFailed)
+		}
+	}
+	return appUUIDStr
+}
+
+
+// getAppNetTable - in 'doAppNet'
+func getAppNetTable(ipaddr string, niStatus *types.NetworkInstanceStatus) {
+	gateway := niStatus.Gateway
+	allIntfs := allUPIntfIPv4()
+	var foundintfName string
+	for _, i := range allIntfs {
+		if i.ipAddr == gateway.String() {
+			foundintfName = i.intfName
+			break
+		}
+	}
+	if foundintfName == "" {
+		fmt.Printf("App ipaddr %s, gateway %v, can't find intf\n", ipaddr, gateway)
+		return
+	}
+
+	link, err := netlink.LinkByName(foundintfName)
+	if err != nil {
+		fmt.Printf("App link get err %v\n", err)
+		return
+	}
+	printColor("\n - ip route tables related to: "+ipaddr, colorGREEN)
+
+	routes := getAllIPv4Routes(link.Attrs().Index)
+	for _, r := range routes {
+		fmt.Printf("%s\n", r.String())
+	}
+}
+
+
+// getVifStats - in 'doAppNet'
+func getVifStats(vifStr string) {
+	retbytes, err := ioutil.ReadFile("/run/zedrouter/NetworkMetrics/global.json")
+	if err != nil {
+		return
+	}
+	printColor("\n - bridge Tx/Rx packets on: "+vifStr, colorGREEN)
+	var ntMetric types.NetworkMetrics
+	_ = json.Unmarshal(retbytes, &ntMetric)
+	for _, m := range ntMetric.MetricList {
+		if vifStr == m.IfName {
+			fmt.Printf(" TxBytes: %d, RxBytes: %d, TxPkts: %d, RxPkts: %d\n",
+				m.TxBytes, m.RxBytes, m.TxPkts, m.RxPkts)
+			break
+		}
+	}
+}
+
+// runAppACLs - in 'doAppNet'
+func runAppACLs(ipStr string) {
+	printColor("\n - check for ACLs on: "+ipStr+"\n", colorGREEN)
+	runAppACLTblAddr("-S", "filter", ipStr)
+	runAppACLTblAddr("-nvL", "filter", ipStr)
+	runAppACLTblAddr("-S", "nat", ipStr)
+	runAppACLTblAddr("-nvL", "nat", ipStr)
+}
+
+func runAppACLTblAddr(op, tbl, ipaddr string) {
+	retStr, err := runCmd("iptables "+op+" -t "+tbl, false)
```

Comment:
To examplify my point about runCmd, it is a lot less error prone to do
   args := []string{"op", "-t", tbl"}
  runCmd("iptables", args)
thus runCmds second argument is of type []string and runCmd does not do a strings.Fields().

Reply:
Ok, let me scan through this.